examples/03_better_tool/ex_better_tool/main.rb

```diff
@@ -0,0 +1,183 @@
+# Copyright 2021 Trimble Inc
+# Licensed under the MIT license
+
+require 'sketchup.rb'
+
+module Examples
+  module BetterTool
+
+    class LineTool
+
+      # Threshold in logical screen pixels for when the mouse is considered to
+      # be dragged.
+      APERTURE = 10
+
+      def activate
+        @mouse_ip = Sketchup::InputPoint.new
+        @picked_first_ip = Sketchup::InputPoint.new
+        @x_down = @y_down = nil
+        update_ui
+      end
+
+      def deactivate(view)
+        view.invalidate
+      end
+
+      def enableVCB?
+        picked_first_point?
+      end
+
+      def resume(view)
+        update_ui
+        view.invalidate
+      end
+
+      def onCancel(reason, view)
+        reset_tool
+        view.invalidate
+      end
+
+      def onMouseMove(flags, x, y, view)
+        if picked_first_point?
+          @mouse_ip.pick(view, x, y, @picked_first_ip)
+          Sketchup.vcb_value = @mouse_ip.position.distance(@picked_first_ip.position)
+        else
+          @mouse_ip.pick(view, x, y)
+        end
+        view.tooltip = @mouse_ip.tooltip if @mouse_ip.valid?
+        view.invalidate
+      end
+
+      def onLButtonDown(flags, x, y, view)
+        @x_down = x
+        @y_down = y
+
+        if picked_first_point? && create_edge > 0
+          # When the user have picked a start point and then picks another point
+          # we create an edge and try to create new faces from that edge.
+          # Like the native tool we reset the tool if it created new faces.
+          reset_tool
+        else
+          # If no face was created we let the user chain new edges to the last
+          # input point.
+          @picked_first_ip.copy!(@mouse_ip)
+        end
+
+        update_ui
+        view.invalidate
+      end
+
+      def onLButtonUp(flags, x, y, view)
+        if (x - @x_down).abs > APERTURE || (y - @y_down).abs > APERTURE
```

Comment:
This checks for distance in a rectangular space around the original pick point. Which means it'll be longer towards the corners. I wonder if it's better to cache X and Y as a `Geom::Point3d` (or `Point2d`) and use `.distance` so the distance check is always equal regardless of direction?

```
if @mouse_down.distance([x, y]) > DRAG_THRESHOLD
  # ...
end
```

I think it reads easier as well.